src/test/kotlin/org/hazelcast/configdsl/Sample.kt

```diff
@@ -65,18 +66,18 @@ fun exhaustiveSample() {
             }
             queryCaches {
                 queryCache("I") {
-                    index("II") {
+                    index(attributes = *arrayOf("II")) {
                         batchSize = 5
                     }
                 }
-                queryCache("III", "IV") {
+                queryCache("III", attributes = *arrayOf("IV")) {
                     bufferSize = 1
                 }
-                queryCache("V", "VI", true)
+                this@map.queryCache("V",IndexType.SORTED, "VI")
```

Comment:
Why prefix with `this@map`?

Reply:
the context of the first two functions is QueryCacheConfigs, but this last function is an extension on MapConfig, so it is a bit confiusing

Reply:
Ok. I would suggest to move it out of of the `queryCaches` block directly under the `map` block.

Reply:
I want to know why do we need the two `queryCache` functions that are extensions on `MapConfig` although we have `queryCaches` which takes `QueryCacheConfigs` which has also the two functions?

Reply:
Honestly, I don't remember... I'd suggest to keep this PR as simple as it is, then you can work further on improvements, etc.

WDYT?

Reply:
okay I will do that